docs/articles/migration/350-400.md

```diff
@@ -0,0 +1,278 @@
+
+- critical bug fixes
+- security fixes
+- Angular major version update
+
+This means that Nebular 3 will be maintained for the next 6 months until Angular 9 release.
+We anticipate 3.6.0 version supporting Angular 8 in the following weeks.
+
+<hr>
+
+## Migration Steps
+
+The following migration steps are required to update:
+
+- Update Angular to version 8 *required*
+- Update Nebular to version 4 *required*
+- Fix imports and component properties *required*
+- Migrate Theme *depends*
+- Migrate to `eva-icons`  *optional*
+
+<hr>
+
+###### 1. Update Angular to version 8
+Nebular 4.0 is based on Angular version 8, thus the very first step is to update the last.
+
+Since Angular has an automatic migration `ng update`  in place, the update will look as follows:
+```sh
+ng update @angular/cli @angular/core
+```
+
+Most of the changes will be done automatically.
+If you have any issues or your setup is more complex - refer to the https://update.angular.io/, select your Angular version and follow the steps
+
+<hr>
+
+###### 2. Update Nebular to version 4
+
+```sh
+ng update @nebular/theme
+```
+
+Additionally, if you use any other Nebular packages, you can add them like this:
+
+```sh
+ng update @nebular/theme @nebular/auth @nebular/security
+```
+
+Since Nebular requires `@angular/cdk`, install the latest version:
+
+```sh
+ng update @angular/cdk
+```
+
+<hr>
+
+###### 3. Fix imports and component properties
+
+Nebular 4.0 unifies a lot of component properties making them more intuitive and easier to use.
+Complete table of changes could be found under [Properties changed in Nebular 4.0.0](https://github.com/akveo/nebular/tree/master/src/framework/theme/350-400-properties-changes.md) doc.
+
+<hr>
+
+###### 4. Migrate Theme
+
+This step depends on your setup. 
+In case when you use a CSS version of styles from `prebuilt/THEME_NAME.css` - no action is required, you can omit this step and move on to the next one.
+
+If you use Nebular Theming or have custom themes, follow the steps below:
+
+
+
+**Replace `/deep/` with `::ng-deep`**
+
+So that it won't error with `dart-sass`, which is now the default Angular sass processor.
+
+**Remove, replace, or temporary comment out removed sass mixins**
+
+`nb-radial-gradient`, `nb-right-gradient`, `nb-hero-STATUS-gradient`. Comment out color sass functions (`tint`, `mix`, `darken`, `lighten`, etc).
+We do this to stop sass throwing errors for being unable to process `undefined` values returned by `nb-theme` for removed and renamed variables.
+At this point, your app should build but won't look good. Let's work on that. 
+
+**Update theme variables**
+
+We suggest the following strategy on migrating to the new theme.
+
+1) Comment out all of the variables changed inside of `nb-register-theme()` call
+
+2) In case you changed global background/border colors, you need to update `color-basic` shades. Default shade for both `light` and `dark` variables looks like this:
```

Comment:
Probably we should refer to shades generator so users without designer could generate it?

Reply:
What kind of generators?